Src/Base/AMReX_TinyProfiler.cpp

```diff
@@ -178,6 +184,12 @@ TinyProfiler::stop () noexcept
                 amrex::Gpu::Device::synchronize();
             }
             nvtxRangePop();
+#endif
+#if defined(AMREX_USE_HIP) && defined(AMREX_USE_ROCTX)
+        if (device_synchronize_around_region) {
```

Comment:
Also we probably should move
```
        if (device_synchronize_around_region) {
            amrex::Gpu::Device::synchronize();
        }
```
outside the ifdefs so that it's called for dpcpp too.